onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -90,19 +92,54 @@ typedef enum { CblasLeft=141, CblasRight=142} CBLAS_SIDE;
 #endif
 
 //
-// Forward declare the thread pool implementation class.
-//
-// N.B. Avoid including ONNX Runtime headers here to keep the dependencies for
-// standalone MLAS test executables smaller.
+// Abstract IThreadPool to remove dependencies for onnxruntime thread pool.
 //
 
-namespace onnxruntime {
-    namespace concurrency {
```

Comment:
I don't like this part. 

The Threadpool class should be a normal class, it can't be virtualized as an interface. This is C++, not Java.

Reply:
What's your suggestions here? The main motivation for this PR is to simplify mlas dependencies to onnxruntime_common. Does @RandySheriffH 's suggestion sound good to you?
> Same question as above - we should not letting mlas, an downstream application, to interfere into upperstream commons. A possible cleaner approach is to virtualize ThreadPool functions such as "SimpleParallelFor..." and "ParallelFor", and letting ThreadPool itself as an interface.